pkg/output/stdout.go

```diff
@@ -167,7 +168,20 @@ func (s *Stdoutput) Progress(status ffuf.Progress) {
 	dur -= mins * time.Minute
 	secs := dur / time.Second
 
-	fmt.Fprintf(os.Stderr, "%s:: Progress: [%d/%d] :: Job [%d/%d] :: %d req/sec :: Duration: [%d:%02d:%02d] :: Errors: %d ::", TERMINAL_CLEAR_LINE, status.ReqCount, status.ReqTotal, status.QueuePos, status.QueueTotal, reqRate, hours, mins, secs, status.ErrorCount)
+	//Create full progress string
+	progressString = fmt.Sprintf("%s:: Progress: [%d/%d] :: Job [%d/%d] :: %d req/sec :: Duration: [%d:%02d:%02d] :: Errors: %d ::", TERMINAL_CLEAR_LINE, status.ReqCount, status.ReqTotal, status.QueuePos, status.QueueTotal, reqRate, hours, mins, secs, status.ErrorCount)
+	if len(progressString)+1 >= s.config.TerminalWidth {
+		//if that progress string plus one is larger than the terminal width then replace the progress string with a shorter one
+		progressString = fmt.Sprintf("%s[%.f%%%%]-[%d/%d]", TERMINAL_CLEAR_LINE, ffuf.Percent(status.ReqCount, status.ReqTotal), status.QueuePos, status.QueueTotal)
+
+		if len(progressString) > s.config.TerminalWidth {
+			//finally if the new minified version of progress bar is too large to fit on the users terminal screen silence the progress
+			s.config.Quiet = true
```

Comment:
I think we should not set the Quiet here, but just return without printing the progress. The quiet mode alters the way the results are printed out as well, and this might not be favorable.